src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -33,6 +35,72 @@ public static class EnumTests
                 @"""MyUInt64Enum"" : " + ulong.MaxValue +
                 @"}";
 
+        private static readonly string s_jsonByteEnum =
+                @"{" +
+                @"""MyByteEnum"" : " + byte.MaxValue +
+                @"}";
+
+
+        public static IEnumerable<object[]> JsonWithInvalidOutOfBoundNumber
+        {
+            get
+            {
+                const string InvalidNumber = "18446744073709551616"; // ulong.MaxValue + 1;
+                foreach(var i in new dynamic[] { InvalidNumber, ulong.MaxValue })
+                {
+                    yield return new object[] { $"{{ \"MyEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : {i} }}" };
+                }
+                foreach(var i in new dynamic[] { InvalidNumber })
+                {
+                    yield return new object[] { $"{{ \"MyUInt64Enum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyUInt64Enum\" : {i} }}" };
+                }
+                foreach(var i in new dynamic[] { -1, (ulong)byte.MaxValue + 2 })
+                {
+                    yield return new object[] { $"{{ \"MyByteEnum\" : \"{i}\" }}" };
+                }
+                foreach(var i in new dynamic[] { -1, (ulong)UInt32.MaxValue + 2 })
+                {
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : \"{i}\" }}" };
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> JsonWithAcceptableInvalidNumber
+        {
+            get
+            {
+                foreach(var (i,j) in new (dynamic, dynamic)[] { (-1, -1), ((ulong)byte.MaxValue + 2, (ulong)UInt32.MaxValue + 2) })
+                {
+                    yield return new object[] { 
+                        $"{{ \"MyByteEnum\" : {i}, \"MyUInt32Enum\" : {j} }}",
+                        (SampleByteEnum)(i % (byte.MaxValue + 1)), (SampleUInt32Enum)(j % ((long)UInt32.MaxValue + 1))
+                    };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonWithAcceptableInvalidNumber))]
+        public static void Parse_JsonWithAcceptableInvalidNumber_Success(string json, SampleByteEnum e1, SampleUInt32Enum e2)
+        {
+            SimpleTestClass result = JsonSerializer.Parse<SimpleTestClass>(json);
+            Assert.Equal(e1, result.MyByteEnum);
+            Assert.Equal(e2, result.MyUInt32Enum);
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonWithInvalidOutOfBoundNumber))]
+        public static void Parse_InvalidOutOfBoundsNumber_Throws(string json)
```

Comment:
Do we throw for OutOfBoundsNumber? I thought that's supported. Isn't this throwing only because the enum value in the JSON payload is a string (rather than  JSON number), which is currently not supported.

In that case, we should update the test and split those two instances up.
1) Out of bounds - valid, and works
2) String enum values - throws, not supported.

Reply:
That's correct. Only throws on the unsupported case, when it is quoted as string

Reply:
> will throw

Can you please file a bug for that (and add the test case - disabled with an active issue attribute pointing to the GitHub issue)?

A straight-forward solution would be to throw for all cases where an unsigned .NET type is being assigned a negative value which means we should special case all unsigned numeric types instead of just Uint64, but wouldn't be consistent with existing behavior and what `Enum.TryParse` does:
https://github.com/dotnet/corefx/blob/ff3ad21651536853d156525f50b296fd883e775e/src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs#L12
https://github.com/dotnet/corefx/blob/ff3ad21651536853d156525f50b296fd883e775e/src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs#L42-L49

Alternatively, we need to handle UInt64 differently somehow (especially outside the normal UInt64 range) since there is no built-in numeric type/Utf8JsonReader API larger than uint64 that can cover the out-of-bounds cases.

Reply:
Thanks for the details. I'll add the issue.